src/Http/ResponseTransformer.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http;
+
+use Cake\Network\Response as CakeResponse;
+use Psr\Http\Message\ResponseInterface as PsrResponse;
+use Zend\Diactoros\CallbackStream;
+use Zend\Diactoros\Response as DiactorosResponse;
+use Zend\Diactoros\Stream;
+
+/**
+ * This class converts PSR7 responses into CakePHP ones and back again.
+ *
+ * By bridging the CakePHP and PSR7 responses together, applications
+ * can be embedded as PSR7 middleware in a fully compatible way.
+ *
+ * @internal
+ */
+class ResponseTransformer
```

Comment:
``Psr7RespsonseTransformer`` would have been an even more fitting name as this class does specifically this.

A ``ResponseTransformer`` could do much more.